esmf_regrid/schemes.py

```diff
@@ -147,11 +158,16 @@ def copy_coords(src_coords, add_method):
 
 
 def _regrid_rectilinear_to_rectilinear__prepare(src_grid_cube, tgt_grid_cube):
-    tgt_x, tgt_y = get_xy_dim_coords(tgt_grid_cube)
-    src_x, src_y = get_xy_dim_coords(src_grid_cube)
-
-    grid_x_dim = src_grid_cube.coord_dims(src_x)[0]
-    grid_y_dim = src_grid_cube.coord_dims(src_y)[0]
+    tgt_x = tgt_grid_cube.coord(axis="x")
+    tgt_y = tgt_grid_cube.coord(axis="y")
+    src_x = src_grid_cube.coord(axis="x")
+    src_y = src_grid_cube.coord(axis="y")
+
+    if len(src_x.points.shape) == 1:
```

Comment:
As I read through this, I wonder if a utility function might make things more readable:

```python
if not is_rectilinear(src_x):  # instead of if len(src_x.points.shape) == 1:
   ...
```
where the util function is something like:
```python
def is_rectilinear(coord) -> bool:
    if hasattr(coord, 'points'):
        shape = coord.points.shape
    else:
        shape = coord.shape
    return len(shape) == 2
````